src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/security/JwtService.kt

```diff
@@ -0,0 +1,65 @@
+package uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.security
+
+import io.jsonwebtoken.ExpiredJwtException
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.SignatureAlgorithm
+import mu.KotlinLogging
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.config.JwtConfig
+import java.security.KeyFactory
+import java.security.PrivateKey
+import java.security.PublicKey
+import java.security.spec.PKCS8EncodedKeySpec
+import java.security.spec.X509EncodedKeySpec
+import java.time.Duration
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.temporal.ChronoUnit
+import java.util.Base64
+import java.util.Date
+import java.util.UUID
+
+private val log = KotlinLogging.logger {}
+
+@Service
+class JwtService(jwtConfig: JwtConfig) {
+
+  private val privateKey: PrivateKey = readPrivateKey(jwtConfig.privateKey)
+  private val publicKey: PublicKey = readPublicKey(jwtConfig.publicKey)
+  private val expiry: Duration = jwtConfig.expiry
+
+  private fun readPrivateKey(privateKeyString: String): PrivateKey =
+    PKCS8EncodedKeySpec(Base64.getDecoder().decode(privateKeyString))
+      .let { KeyFactory.getInstance("RSA").generatePrivate(it) }
+
+  private fun readPublicKey(publicKeyString: String): PublicKey =
+    X509EncodedKeySpec(Base64.getDecoder().decode(publicKeyString))
+      .let { KeyFactory.getInstance("RSA").generatePublic(it) }
+
+  fun generateToken(email: String): String =
+    Jwts.builder()
+      .setId(UUID.randomUUID().toString())
+      .setSubject(email)
+      .setExpiration(
+        Date.from(
+          LocalDateTime.now().plus(expiry.toMillis(), ChronoUnit.MILLIS).toInstant(zoneOffset())
```

Comment:
Minor, but could we simplify this a little? We are generating an `Instant` via a `LocalDateTime` - I've not tried, but would this work and be simpler?
```
Date.from(
  Instant.now().plus(expiry.toMillis(), ChronoUnit.MILLIS)
)
```

Reply:
Agreed - I'll trry and simplify